go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/rook/rook
 
-go 1.20
+go 1.23.0
+
+toolchain go1.23.1
```

Comment:
I think the Rook makefile will complain about this even for our downstream builds. We could make a downstream patch to allow 1.23 in the makefile, but we will also have to coordinate with the build team.

I know we had a group meeting a while back on this topic, but I don't recall what our specific guidance is for this scenario. Let's reach out to Sunil to understand better.